Code/Framework/AzCore/AzCore/Console/Console.cpp

```diff
@@ -484,9 +484,23 @@ namespace AZ
                     result = true;
                     if ((silentMode == ConsoleSilentMode::NotSilent) && (curr->GetFlags() & ConsoleFunctorFlags::IsInvisible) != ConsoleFunctorFlags::IsInvisible)
                     {
-                        CVarFixedString value;
-                        curr->GetValue(value);
-                        AZLOG_INFO("> %s : %s", curr->GetName(), value.empty() ? "<empty>" : value.c_str());
+                        // First use the ConsoleFunctorBase::GetValue function
+                        // to retrieve the value of the type of the first template parameter to the ConsoleFunctor class template
+                        // This is populated for non-void types and is set for Console Variables(CVars) and Console Commands
+                        // which are member functions
+                        // See `ConsoleFunctor<_TYPE, _REPLICATES_VALUE>::GetValueAsString`
+                        CVarFixedString inputStr;
+                        if (GetValueResult getCVarValue = curr->GetValue(inputStr);
+                            getCVarValue != GetValueResult::Success)
+                        {
+                            // In this case the ConsoleFunctorBase pointer references a `ConsoleFunctor<void, _REPLICATES_VALUE>` object
+                            // which has no type associated with it.
+                            // This is used for Console Commands which are free functions
+                            // The `ConsoleFunctor<void, _REPLICATES_VALUE>::GetValueAsString` returns NotImplemented
```

Comment:
It's been a long time, but the _REPLICATES_VALUE template parameter is always false in this case, no?
What happens if I enter complete garbage after the command name? It looks like this falls back to a pure echo if the GetValue() call fails.

Reply:
_REPLICATES_VALUE is actually hardcoded to true for [Console Variables](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.h#L127)

> What happens if I enter complete garbage after the command name?
* Well if the command is a console variable, then it would output the value of it. Because the string argument wouldn't be convertible to a type that isn't ConsoleFunctor can store, the old value would be output.
* If the command is a console **free** function, then it would just output the the input arguments the user passed in.
* If the command is a console **member** function, then it would output a string converted from the class type associated with the member function.